feat: add CAFxX/mgo

iwata opened the pull request:
[CAFxX/mgo](https://github.com/CAFxX/mgo): Build and bundle multiple GOAMD64 variants in a single executable

suzuki-shunsuke commented:
Thank you!

suzuki-shunsuke commented on pkgs/CAFxX/mgo/registry.yaml:
```diff
@@ -0,0 +1,8 @@
+    repo_owner: CAFxX
+    repo_name: mgo
+    description: Build and bundle multiple GOAMD64 variants in a single executable
+    supported_envs:
+      - darwin
+      - linux
```
I'm not familiar with this tool, but this tool doesn't work on Windows?

suzuki-shunsuke replied:
```console
$ mgo --help
INFO[0000] Installing a Go tool                          aqua_version=2.21.0 env=darwin/arm64 exe_name=mgo go_package_path=github.com/CAFxX/mgo@v0.2.12 gobin=/Users/shunsukesuzuki/.local/share/aquaproj-aqua/pkgs/go_install/github.com/CAFxX/mgo/v0.2.12/bin package_name=CAFxX/mgo package_version=v0.2.12 program=aqua registry=standard
go: downloading github.com/CAFxX/mgo v0.2.12
go: downloading golang.org/x/sync v0.2.0
go: downloading golang.org/x/mod v0.10.0
GOOS="darwin" is not supported
```

Oh? `darwin` isn't supported?

suzuki-shunsuke replied:
> Currently only GOOS=linux and GOARCH=amd64 are supported, and only in buildmodes that produce executables (not archives, plugins, or libraries)

https://github.com/CAFxX/mgo/blob/8e4a2e5f301ba679868b70fdc27ae7cbd47b806a/README.md?plain=1#L50-L51

iwata replied:
Yes, but should I limit it to `linux/amd64` only?

![CleanShot 2023-12-13 at 07 51 40@2x](https://github.com/aquaproj/aqua-registry/assets/121048/19689be5-a5dd-4833-a5e1-4f18789b8e9e)
https://github.com/CAFxX/mgo#notes

suzuki-shunsuke replied:
Yeah, I think so.

iwata replied:
fixed `supported_envs`

iwata committed:
feat: add CAFxX/mgo

[CAFxX/mgo](https://github.com/CAFxX/mgo): Build and bundle multiple GOAMD64 variants in a single executable

suzuki-shunsuke commented:
v4.103.0 is out 🎉
https://github.com/aquaproj/aqua-registry/releases/tag/v4.103.0
